### DISCUSSION TITLE
Render deploy error

### DISCUSSION THREAD

**Discussion by lfv89** [AllYourBot/hostedgpt#322] [General]
Hi, I'm getting this error when using the `Render` deploy button:

```
No open ports detected, continuing to scan...
==> Docs on specifying a port: https://render.com/docs/web-services#port-binding
=> Booting Puma
=> Rails 7.1.3.2 application starting in production 
=> Run `bin/rails server --help` for more startup options
I, [2024-05-04T17:14:09.393672 #1]  INFO -- : Configuring active record encryption from environment
Exiting
/app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `exec': PG::UndefinedTable: ERROR:  relation "active_storage_files" does not exist (ActiveRecord::StatementInvalid)
LINE 10:  WHERE a.attrelid = '"active_storage_files"'::regclass
                             ^
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:19:in `block (2 levels) in query'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1028:in `block in with_raw_connection'
	from /app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/concurrency/null_lock.rb:9:in `synchronize'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1000:in `with_raw_connection'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:18:in `block in query'
	from /app/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.3.2/lib/active_support/notifications/instrumenter.rb:58:in `instrument'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract_adapter.rb:1143:in `log'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:17:in `query'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/postgresql_adapter.rb:1074:in `column_definitions'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:109:in `columns'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:346:in `block in columns'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:345:in `fetch'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:345:in `columns'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:354:in `block in columns_hash'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:353:in `fetch'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:353:in `columns_hash'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:58:in `columns_hash'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/connection_adapters/schema_cache.rb:188:in `columns_hash'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/model_schema.rb:618:in `load_schema!'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/attributes.rb:264:in `load_schema!'
	from /app/vendor/bundle/ruby/3.2.0/gems/activerecord-7.1.3.2/lib/active_record/encryption/encryptable_record.rb:127:in `load_schema!'
```

**Comment by keithschacht** [comment]
Oh shoot, I just refactored a bunch of the dockerfile configuration over the last few days but I thought I tested all the scenarios before merging in! Thank you for reporting this, let me investigate and get this fixed.

**Comment by keithschacht** [comment]
@lfv89 In a test environment I deleted all Render services, updated my hostedgpt fork to make sure it had the latest commits, and I clicked the Render button. Everything worked for me.

The one clue that I see in your error message is that the database tables failed to be created. This is either because the DB never got setup, or maybe the dB did get setup but the Web Service failed to run a DB setup script failed. But I don’t have any ideas yet as to why this would happen for you but not in the test I just did. I have not heard of any other people experiencing your issue.

I need some more information to help debug this:

(1) First, can you confirm that your fork is up to date? (and let me know the URL to your github fork)
(2) Can you confirm that on your Render dashboard you have 3 services: a web service, postgres, and redis. Which are red and which are green?
(3) If the only service that is red is Web Service, maybe this failure was a random Render incident. Go to your Render dashboard, click the web service, click the Manual Deploy button (top right) and “deploy the latest commit”. But if there are other failed services then I’ll wait to hear details on those.

**Comment by lfv89** [comment]
Hi, 

I'm not sure what happened, but I just deleted every service from Render, "deployed" again and this time I chose "Update Blueprint" instead of "Create Blueprint" as I was doing previously, and it works now!

**Reply by keithschacht** [reply]
Oh good to hear! Thanks for updating me on this.

**Comment by Pinguu-dev** [comment]
I have the same issue here @krschacht 

Fresh Fork.

**Reply by keithschacht** [reply]
Hi @Pinguu-dev can you go to the logs during the deploy and copy & paste any error messages that you're getting?

**Reply by keithschacht** [reply]
@Pinguu-dev I was able to repro the Render issue; I broke it with a recent commit. I just pushed a fix for this and I confirmed I was able to deploy a fresh Render. Go ahead and give it a try again. Make sure your fork is sync'd to pull in that latest commit, and then in Render if it does not automatically deploy then click on your Web Service, click "Manual Deploy" in the top right, and click "Deploy latest commit."

Sorry about the trouble! But I really appreciate you reporting this so I could get it fixed quickly. Let me know if this works for you.
